AK/CircularBuffer.h

```diff
@@ -41,6 +41,23 @@ class CircularBuffer {
 
     void clear();
 
+    template<typename StreamType>
```

Comment:
Maybe we can move the `StreamLike` from `BufferedStream` to `Stream.h` and use it here.

Reply:
Actually this made me realize this doesn't need to be a template at all